File: backend-common/src/main/java/cc/jktu/apps/common/config/TimeMetaObjectHandler.java
package cc.jktu.apps.common.config;

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler;
import org.apache.ibatis.reflection.MetaObject;

import java.time.Instant;

public class TimeMetaObjectHandler implements MetaObjectHandler {

    @Override
    public void insertFill(MetaObject metaObject) {
        strictInsertFill(metaObject, "createTime", () -> Instant.now().getEpochSecond(), Long.class);
        strictInsertFill(metaObject, "updateTime", () -> Instant.now().getEpochSecond(), Long.class);
    }

    @Override
    public void updateFill(MetaObject metaObject) {
        strictUpdateFill(metaObject, "updateTime", () -> Instant.now().getEpochSecond(), Long.class);
    }

}

File: backend-common/src/main/kotlin/cc/jktu/apps/common/config/TimeMetaObjectHandler.kt
package cc.jktu.apps.common.config

import com.baomidou.mybatisplus.core.handlers.MetaObjectHandler
import org.apache.ibatis.reflection.MetaObject
import java.time.Instant

class TimeMetaObjectHandler : MetaObjectHandler {

    override fun insertFill(metaObject: MetaObject) {
        strictInsertFill(metaObject, "createTime", { Instant.now().epochSecond }, Long::class.java)
        strictInsertFill(metaObject, "updateTime", { Instant.now().epochSecond }, Long::class.java)
    }

    override fun updateFill(metaObject: MetaObject) {
        strictUpdateFill(metaObject, "updateTime", { Instant.now().epochSecond }, Long::class.java)
    }
}
